packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -141,11 +146,28 @@ export const fetchImageSource = async (src: string, credential: CommunicationTok
     const headers = new Headers();
     headers.append('Authorization', `Bearer ${token}`);
     try {
-      return await fetch(url, { headers });
+      return await fetchWithTimeout(url, { headers });
     } catch (err) {
       throw new ChatError('ChatThreadClient.getMessage', err as Error);
     }
   }
+  async function fetchWithTimeout(
+    resource: string | URL | Request,
+    options: { timeout?: number; headers?: Headers }
+  ): Promise<Response> {
+    // default timeout is 4 seconds
```

Comment:
4 seconds seems very short?
Where did this value originate?

Reply:
Do we have a time out value somewhere

Reply:
Image sizes can be anywhere from tens of kb to multi-megabytes. 
If any operation takes > 4 seconds, it will fail. 
Networks can be unexpectedly slow, and the fetch doesn't have a retry mechanism at this point IIRC.

Reply:
true, what's the right timeout then, would 10 seconds be reasonable?

Reply:
Is the purpose of the time out to work around lack of browser support for detecting network disconnect and therefore a failed download?

What would be a better way to handle this?

Reply:
it's nothing to do with the browser itself. When we make a request, we are responsible to cancel it if it's taking too long. Unlike in `jQuery` you can define a timeout in `Ajax` request directly, `fetch` currently doesn't have it (https://fetch.spec.whatwg.org/#connections). So we need to leverage AbortController to cancel async requests: https://developer.mozilla.org/en-US/docs/Web/API/AbortController/signal

Reply:
also this is a pretty standard way to implement timeout with fetch, if you google "fetch timeout" then you will find pretty much the same solutions

Reply:
I was saying that for fetch, it doesn't have API for timeout by default (unlike ajax), we need to leverage AbortController to kill the operation. And in the doc you linked, they has the same approach: https://betterstack.com/community/guides/scaling-nodejs/nodejs-timeouts/#timing-out-a-fetch-api-request

<img width="799" alt="image" src="https://github.com/Azure/communication-ui-library/assets/109105353/2980f0b3-644a-496d-a605-c831becda45a">

30 seconds sound good to me since it matches the AFD timeout which is 30s as well

Reply:
checked with Emlyn offline to confirm that 30 seconds are reasonable and having a timeout callback is justified so that we can have a consistent behaviour in different browsers